### DISCUSSION TITLE
Invert default selection from all to empty in selectors with multiple activated

### DISCUSSION THREAD

**Discussion by shaunwbell** [Avaiga/taipy#592] [Q&A] [answered]
Is there a way to invert the default selections from *all* to empty when using a selector with multiple ?

I would like to build a filter that works by starting with no constraints and then applying new ones, rather than the other way around (all options selected by default).  The list I would be putting into the dropdown is fairly large and I would rather select the one or two options I want at any given time as opposed to remove the 10's of options I don't.

My code snippet to build the selector is as follows

`
mooring_id = mooring_ids = list(df["MooringID"].unique()) #list of >100 options

<|{mooring_id}|selector|lov={mooring_ids}|multiple|label=Select the Mooring ID|dropdown|on_change=on_filter|class_name=fullwidth|>
`

**Comment by FlorianJacta** [comment, marked as answer]
You have initialized your selection to be all the selector values in your code. You can initialize it to None to solve your issue:

```python
from taipy.gui import Gui 

mooring_ids = [f"Choice {i}" for i in range(100)]
mooring_id = None

md = "<|{mooring_id}|selector|lov={mooring_ids}|multiple|label=Select the Mooring ID|dropdown|on_change=on_filter|class_name=fullwidth|>"

def on_filter(state):
    print(state.mooring_id)

Gui(md).run()
```

**Comment by shaunwbell** [comment]
I did have to put it in an empty list but otherwise this is exactly what I wanted.  Thanks

```python
mooring_id = None
```
became
```python
mooring_id = [None]
```

**Reply by FlorianJacta** [reply]
I guess you could just put `mooring_id = []` then.

**Reply by DeveloperVaibhav1** [reply]
> I did have to put it in an empty list but otherwise this is exactly what I wanted. Thanks
> 
> ```python
> mooring_id = None
> ```
> 
> became
> 
> ```python
> mooring_id = [None]
> ```

I guess there would be mooring id = [] then its correct sintax according to me

**Reply by DeveloperVaibhav1** [reply]
Hello. The template tests are designed to run the application in a subprocess after being created by the template, and wait for a timeout to return the result from the subprocess.
The default timeout is 30 seconds, which is defined in the tests.templates.utils._run_template().

You can try to reduce the timeout to make the test faster on a local machine since running the application should be pretty quick.
However, please don't commit the change of the timeout since it takes much slower to run the tests on Github Action on a subprocess, and 30s is reasonable to make sure the tests pass.
